xamarin/Xamarin.Forms
# Set agent os for all pools

### Description of Change ###

- Set the OS demand for all the pools so that everything uses the correct operating system to build. 
- Added conditional windows steps so it would use demands when using our bots opposed to pipelines
- added a check for xcode version so it'll install the correct sdks

### Testing Procedure ###
- Turn on the bots that are confusing the builds and make sure everything still builds

Internal
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3642726&view=results

Hosted 10.15
https://dev.azure.com/xamarin/public/_build/results?buildId=16940&view=logs&j=be1d3261-c4d2-5528-aad8-51398c05403f

Bot turned on with 10.15
https://dev.azure.com/xamarin/public/_build/results?buildId=16939&view=results

Hosted 10.14 (normal run without modifying parameters)
https://dev.azure.com/xamarin/public/_build/results?buildId=16938&view=logs&j=1c8381ff-b205-54ef-ec4d-2dac1b85342d

### PR Checklist ###
